packages/app-store/_utils/getAppCategories.ts

```diff
@@ -1,10 +1,13 @@
+import { WEBAPP_URL } from "@calcom/lib/constants";
 import { Icon } from "@calcom/ui";
 
 function getHref(baseURL: string, category: string, useQueryParam: boolean) {
-  return useQueryParam ? `${baseURL}&category=${category}` : `${baseURL}/${category}`;
+  const baseUrlParsed = new URL(baseURL, WEBAPP_URL);
+  baseUrlParsed.searchParams.set("category", category);
```

Comment:
&category can only be suffixed if there is a query param already. So, handle it in a better way